### PR TITLE
Add tt create, build

### LINKED ISSUE #2564: [2pt] tt create
**Epic: [Document TT CLI](https://github.com/tarantool/doc/issues/2553)**
Related dev issue: https://github.com/tarantool/tt/issues/12
Related dev epic: https://github.com/tarantool/roadmap-internal/issues/125

Product: Tarantool CLI
Since: 2.10
Audience/target: dev, admin
Root document: TBA
SME: @ Totktonada

### Details

The result may look similar to the `cartridge create` [documentation](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_cli/commands/create/).

### LINKED ISSUE #2999: tt build
**Epic: [Document TT CLI](https://github.com/tarantool/doc/issues/2553)**
Related dev issue: https://github.com/tarantool/tt/issues/40
Related dev epic: https://github.com/tarantool/roadmap-internal/issues/125

Product: Tarantool CLI
Since: 2.10
Audience/target: dev, admin
Root document: TBA
SME: @ LeonidVas

### Details

The result may look similar to the `cartridge build` [documentation](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_cli/commands/build/).

### PR DESCRIPTION
**Pull request by p7nov**
Resolves #2564
Resolves #2999

### CODE REVIEW

File: doc/reference/tooling/tt_cli/commands.rst [hunk 1]
```diff
@@ -24,6 +26,8 @@ help for the given command.
             -   Connect to a Tarantool instance
         *   -   :doc:`coredump <coredump>`
             -   Manipulate Tarantool core dumps
+        *   -   :doc:`create <create>`
+            -   Manipulate Tarantool core dumps
```
Comment (psergee) [inline doc/reference/tooling/tt_cli/commands.rst]: This description is from other command.

**Review by psergee — APPROVED** [review]

File: doc/reference/tooling/tt_cli/build.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Building an application
+====================
+
+..  code-block:: bash
+
+    tt build [PATH] [flags]
+
+
+``tt build `` builds a Tarantool application locally.
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/build.rst]:
```suggestion
``tt build`` builds a Tarantool application locally.
```
Typo

File: doc/reference/tooling/tt_cli/build.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+Details
+-------
+
+The ``PATH`` argument should contain the path to the application directory
+(that is, to the build source). The default path is ``.`` (current directory).
+
+The application directory must contain a ``.rockspec`` file to use for the build.
+If there are more than one ``.rockspec`` files in the application directory, specify
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/build.rst]:
```suggestion
If there is more than one ``.rockspec`` file in the application directory, specify
```

File: doc/reference/tooling/tt_cli/build.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+and located in the application directory.
+
+.. note::
+
+    For compatibility with Cartridge applications, the pre-build and post-build scripts
+    can also have names ``cartridge.pre-build`` and ``cartridge.post-build``.
+
+If your application depends on closed-source rocks, or if the build should contain
+rocks from a project added as a submodule, **install** all these
+dependencies **before** building using the pre-build script.
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/build.rst]: The phrase is ambiguous, and the connection between words is not quite clear. Do you mean "install using the pre-build script" or "building using the pre-build script"?

File: doc/reference/tooling/tt_cli/create.rst [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+            -   Variable definition. Usage: ``--var var_name=value``
+        *   -   ``--vars-file``
+            -   Path to the file with variable definitions
+
+Details
+-------
+
+*Application templates* speed up the development of Tarantool applications by
+defining their initial structure and content. A template can include the application
+code, configuration, build scripts, and other resources.
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/create.rst]:
```suggestion
defining their initial structure and content. A template can include application
code, configuration, build scripts, and other resources.
```
A template is a blueprint for "some" application, so I suggest removing the article.

File: doc/reference/tooling/tt_cli/create.rst [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+``tt`` searches templates in the directories specified in the ``templates`` section
+of its :ref:`configuration file <tt-config_file>`.
+
+Template structure
+~~~~~~~~~~~~~~~~~~
+
+Application templates are directories with files.
+
+The main file of a template is its *manifest*. It defines how the applications
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/create.rst:58]: Is there a convention on how to name such a manifest file?

File: doc/reference/tooling/tt_cli/create.rst [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+A template manifest is a YAML file that can contain the following sections:
+
+*   ``description`` -- the template description
+*   ``vars`` -- :ref:`template variables <template-variables>`
+*   ``pre-hook`` and ``post-hook`` -- paths to executables to run before and after the template
+    instantiation
+*   ``include`` -- a list of files to keep in the application directory after
+    instantiation. If this section is omitted, the application will contain all template files
+    and directories.
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/create.rst]:
```suggestion
*   ``description`` -- the template description.
*   ``vars`` -- :ref:`template variables <template-variables>`.
*   ``pre-hook`` and ``post-hook`` -- paths to executables to run before and after the template
    instantiation.
*   ``include`` -- a list of files to keep in the application directory after
    instantiation. If this section is omitted, the application will contain all template files
    and directories.
```
Use consistent ending punctuation. See our [style guide](https://www.tarantool.io/en/doc/latest/contributing/docs/style/#ending-punctuation-in-lists-and-tables).

File: doc/reference/tooling/tt_cli/create.rst [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+      - prompt: Retry count
+        default: "3"
+        name: retry_count
+        re: ^\d+$
+    pre-hook: ./hooks/pre-gen.sh
+    post-hook: ./hooks/post-gen.sh
+    include:
+    - init.lua
+    - instances.yml
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/create.rst]:
```suggestion
      - init.lua
      - instances.yml
```
I think this is the proper YAML syntax.

File: doc/reference/tooling/tt_cli/create.rst [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+values and the ``*.tt.template`` extension is removed.
+
+.. _template-variables:
+
+Variables
+~~~~~~~~~
+
+Templates variables are replaced with their values provided upon the instantiation.
+
+All templates have the ``name`` variable. Its value is taken from the `--name`` flag.
```
Comment (patiencedaur) [inline doc/reference/tooling/tt_cli/create.rst]:
```suggestion
All templates have the ``name`` variable. Its value is taken from the ``--name`` flag.
```
Typo

**Review by patiencedaur — COMMENTED** [review]
Thanks! Here are some suggestions.

**Pull request merged**